router/rate_limiting.go

```diff
@@ -0,0 +1,109 @@
+package router
+
+import (
+	"container/list"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/topfreegames/pitaya/cluster"
+	"github.com/topfreegames/pitaya/config"
+	"github.com/topfreegames/pitaya/constants"
+	"github.com/topfreegames/pitaya/errors"
+	"github.com/topfreegames/pitaya/logger"
+	"github.com/topfreegames/pitaya/route"
+	"github.com/topfreegames/pitaya/session"
+)
+
+// WithRateLimiting wraps routingFunc and controls
+// player rate through session.
+func WithRateLimiting(c *config.Config, f RoutingFunc) RoutingFunc {
+	limit := c.GetInt("pitaya.router.ratelimiting.limit")
+	interval := c.GetDuration("pitaya.router.ratelimiting.interval")
+	forceDisable := c.GetBool("pitaya.router.ratelimiting.forcedisable")
+	if forceDisable {
+		logger.Log.Warn("rate limiting is forcibly disabled")
+	}
+
+	return func(
+		ctx context.Context,
+		r *route.Route,
+		payload []byte,
+		servers map[string]*cluster.Server,
+	) (*cluster.Server, error) {
+		if forceDisable {
```

Comment:
this is an optional feature that one had intentionally included, is the point of having this config just to make it easier to turn it off without having to make a new build?

Reply:
Yes. In case you notice that players are being blocked by rate limiting unintentionally and you want to disable it for now to test other values for limit and interval